solarforecastarbiter/io/reference_observations/arm.py

```diff
@@ -176,3 +231,138 @@ def update_observation_data(api, sites, observations, start, end):
             api, partial(fetch, doe_arm_user_id=doe_arm_user_id,
                          doe_arm_api_key=doe_arm_api_key),
             site, observations, start, end)
+
+
+def adjust_site_parameters(site):
+    """Updates extra parameters with applicable datastreams from
+    `arm_reference_sites.json`
+
+    Parameters
+    ----------
+    site: dict
+
+    Returns
+    -------
+    dict
+        Copy of input with updated extra parameters.
+    """
+    with open(DEFAULT_SITEFILE) as fp:
+        sites_metadata = json.load(fp)['sites']
+
+    # ARM has multiple 'locations' at each 'site'. For example, the Southern
+    # Great Plains (SGP) ARM site has many locations throughout Oklahoma, and
+    # neighboring states. Each location is identified by a code, e.g. Byron
+    # Oklahoma is location `E11` at the SGP site, and it's data is accessed via
+    # datastreams with the pattern `sgp<product>e11.<data-level>` where product
+    # indicates the contents of the datastream (we are interested in `met` and
+    # `qcrad1long` products) and data-level indicates quality and any
+    # processing applied to the data. In the Solar Forecast Arbiter we store
+    # each 'location' as a SFA site. We use the `network_api_id` to indicate
+    # the ARM site's location code and `network_api_abbreviation` to represent
+    # ARM "site code (e.g. `sgp`). Both of these keys must be used to  identify
+    # a site in the Solar Forecast Arbiter, because ARM's location ids are only
+    # unique for a given ARM site.
+    arm_location_id = site['extra_parameters']['network_api_id']
+    arm_site_id = site['extra_parameters']['network_api_abbreviation']
+
+    for site_metadata in sites_metadata:
+        site_extra_params = json.loads(site_metadata['extra_parameters'])
+        if (
+            site_extra_params['network_api_id'] == arm_location_id
+            and site_extra_params['network_api_abbreviation'] == arm_site_id
+
+        ):
+            site_out = site.copy()
+            site_out['extra_parameters'] = site_extra_params
+            return site_out
+    return site
+
+
+def find_stream_data_availability(streams, start, end):
+    """Determines what date ranges to use for each datastream. Date ranges of
+    each stream must not overlap.
```

Comment:
I think this should raise an error if they do overlap. Just to be certain of the behaviour and so we catch invalid params